pkg/validate/trigger.go

```diff
@@ -0,0 +1,121 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package validate
+
+import (
+	"context"
+	"fmt"
+
+	build "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/utils/pointer"
+)
+
+// Trigger implements the interface BuildPath with the objective of applying validations against the
+// `.spec.trigger` related attributes.
+type Trigger struct {
+	build *build.Build // build instance
+}
+
+// validate goes through the trigger "when" conditions to validate each entry.
+func (t *Trigger) validate(triggerWhen []build.TriggerWhen) []error {
+	var allErrs []error
+	for _, when := range triggerWhen {
+		if when.Name == "" {
+			t.build.Status.Reason = build.BuildReasonPtr(build.TriggersNameCanNotBeBlank)
+			t.build.Status.Message = pointer.String("name is not set on when trigger condition")
+			allErrs = append(allErrs, fmt.Errorf("%s", *t.build.Status.Message))
+		}
+
+		switch when.Type {
+		case build.WhenTypeGitHub:
+			if when.GitHub == nil {
+				t.build.Status.Reason = build.BuildReasonPtr(build.TriggersInvalidGitHub)
+				t.build.Status.Message = pointer.String(fmt.Sprintf(
+					"%q is missing required attribute `.github`", when.Name,
+				))
+				allErrs = append(allErrs, fmt.Errorf("%s", *t.build.Status.Message))
+			} else {
+				if len(when.GitHub.Events) == 0 {
```

Comment:
wondering if this is needed, as I saw somewhere else a kubebuilder annotation on `+kubebuilder:validation:MinLength` , any ideas?

Reply:
The KubeBuilder annotation `+kubebuilder:validation:MinItems=1` works on the API validation level making sure the request returns error when `MinItems` is not reached. However, code wise we also need to catch this scenario and make the business logic works as expected. Those are complementary measurements.